immunopepper/io_.py

```diff
@@ -1,204 +1,215 @@
-       filepointer.kmer_segm_expr_fp['pqwriter'] = save_pd_toparquet(segm_path, data[1],
-                         compression=compression, verbose=verbose, pqwriter=filepointer.kmer_segm_expr_fp['pqwriter'], writer_close=False)
-       filepointer.kmer_edge_expr_fp['pqwriter'] = save_pd_toparquet(edge_path, data[2],
-                         compression=compression, verbose=verbose, pqwriter=filepointer.kmer_edge_expr_fp['pqwriter'], writer_close=False)
-
-
-def initialize_fp(output_path, mutation_mode, gzip_tag,
-                  kmer_list, output_fasta, cross_graph_expr):
-
-    ### Paths
-    background_peptide_file_path = os.path.join(output_path, mutation_mode+ '_annot_peptides.fa.pq' + gzip_tag)
-    background_kmer_file_path = os.path.join(output_path, mutation_mode + '_annot_kmer.pq' + gzip_tag)
-    gene_expr_file_path = os.path.join(output_path, 'gene_expression_detail.pq' + gzip_tag)
+def save_kmer_matrix(data, graph_samples, filepointer: Filepointer, compression: str = None, out_dir: str = None,
+                     verbose: bool = False):
+    """
+    Saves the kmer segment and junction matrices per gene.
+    The filepointer is kept open until all genes from the batch are written
+    """
+    if data[0]:
+        segm_path = get_save_path(filepointer.kmer_segm_expr_fp, out_dir)
+        edge_path = get_save_path(filepointer.kmer_edge_expr_fp, out_dir)
+        data[1] = pd.DataFrame.from_dict(data[1], orient='index', columns=graph_samples).reset_index().rename(
+            {'index': filepointer.kmer_segm_expr_fp['columns'][0]}, axis=1)
+        data[2] = pd.DataFrame.from_dict(data[2], orient='index', columns=graph_samples).reset_index().rename(
+            {'index': filepointer.kmer_segm_expr_fp['columns'][0]}, axis=1)
+
+        data[1][filepointer.kmer_segm_expr_fp['columns'][1]] = data[0].values()
+        data[2][filepointer.kmer_segm_expr_fp['columns'][1]] = data[0].values()
+        filepointer.kmer_segm_expr_fp['pqwriter'] = save_pd_toparquet(segm_path, data[1],
+                                                                      compression=compression, verbose=verbose,
+                                                                      pqwriter=filepointer.kmer_segm_expr_fp[
+                                                                          'pqwriter'], writer_close=False)
+        filepointer.kmer_edge_expr_fp['pqwriter'] = save_pd_toparquet(edge_path, data[2],
+                                                                      compression=compression, verbose=verbose,
+                                                                      pqwriter=filepointer.kmer_edge_expr_fp[
+                                                                          'pqwriter'], writer_close=False)
+
+
+def initialize_fp(output_path: str, mutation_mode: str,
+                  kmer_list: list[int], output_fasta: bool, cross_graph_expr: bool):
+    """"
```

Comment:
Just double-checking. The 'gzip_tag' seems to have been dropped. Compression is no longer needed or just not reflected in the filename?

Reply:
Yeah, the function was called in only one place, and the gzip tag was set to the empty string, so I simply removed the parameter. No functionality was changed. I believe that a file is automatically gzipped if it ends with `gz` in the name (see `open_gz_or_normal` at line 28, but at least in build mode this never happens.